Mathlib/Algebra/Star/CentroidHom.lean

```diff
@@ -0,0 +1,144 @@
+  toNonUnitalRingHom :=
+    (SubsemiringClass.subtype (Subsemiring.center (CentroidHom α))).toNonUnitalRingHom
+  map_star' f := by
+    simp only [RingHom.toMonoidHom_eq_coe, OneHom.toFun_eq_coe, MonoidHom.toOneHom_coe,
+      MonoidHom.coe_coe, SubsemiringClass.coe_subtype]
+    exact rfl
+
+theorem star_centerToCentroidCenter (z : NonUnitalStarSubsemiring.center α) :
+    star (centerToCentroidCenter z) =
+      (centerToCentroidCenter (star z : NonUnitalStarSubsemiring.center α)) := by
+  ext a
+  calc
+      (star (centerToCentroidCenter z)) a = star (z * star a) := rfl
+      _ = star (star a) * star z := by simp only [star_mul, star_star, StarMemClass.coe_star]
+      _ = a * star z := by rw [star_star]
+      _ = (star z) * a := by rw [(star z).property.comm]
+      _ = (centerToCentroidCenter ((star z) : NonUnitalStarSubsemiring.center α)) a := rfl
+
+/-- The canonical *-homomorphism from the center of a non-unital, non-associative *-semiring into
+the center of its centroid. -/
+def starCenterToCentroidCenter :
+    NonUnitalStarSubsemiring.center α →⋆ₙ+* Subsemiring.center (CentroidHom α) where
+  toNonUnitalRingHom := centerToCentroidCenter
+  map_star' _ := by
+    simp only [MulHom.toFun_eq_coe, NonUnitalRingHom.coe_toMulHom]
+    exact (star_centerToCentroidCenter _).symm
+
+/-- The canonical homomorphism from the center into the centroid -/
+def starCenterToCentroid : NonUnitalStarSubsemiring.center α →⋆ₙ+* CentroidHom α :=
+  NonUnitalStarRingHom.comp (centerStarEmbedding) (starCenterToCentroidCenter)
+
+lemma starCenterToCentroid_apply (z : NonUnitalStarSubsemiring.center α) (a : α) :
+    (starCenterToCentroid z) a = z * a := rfl
+
+/--
+Let `α` be a star ring with commutative centroid. Then the centroid is a star ring.
+-/
+@[reducible]
+def starRingOfCommCentroidHom (mul_comm : Std.Commutative (α := CentroidHom α) (· * ·)) :
+    StarRing (CentroidHom α) where
+  __ := instStarAddMonoid
+  star_mul _ _ := ext (fun _ => by
+    rw [mul_comm.comm, star_apply, mul_apply, mul_apply, star_apply, star_apply, star_star])
+
+end NonUnitalNonAssocStarSemiring
+
+section NonAssocStarSemiring
+
+variable [NonAssocSemiring α] [StarRing α]
+
+/-- The canonical isomorphism from the center of a (non-associative) semiring onto its centroid. -/
+def starCenterIsoCentroid : StarSubsemiring.center α ≃⋆+* CentroidHom α where
+  __ := starCenterToCentroid
+  invFun T :=
+    ⟨T 1, by refine ⟨?_, ?_, ?_, ?_⟩; all_goals simp [← map_mul_left, ← map_mul_right]⟩
+  left_inv z := Subtype.ext <| by simp only [MulHom.toFun_eq_coe,
+    NonUnitalRingHom.coe_toMulHom, NonUnitalStarRingHom.coe_toNonUnitalRingHom,
+    starCenterToCentroid_apply, mul_one]
+  right_inv T := CentroidHom.ext <| fun _ => by
+    simp [starCenterToCentroid_apply, ← map_mul_right]
+
+@[simp]
+lemma starCenterIsoCentroid_apply (a : ↥(NonUnitalStarSubsemiring.center α)) :
+    CentroidHom.starCenterIsoCentroid a = CentroidHom.starCenterToCentroid.toFun a := rfl
```

Comment:
```suggestion
    CentroidHom.starCenterIsoCentroid a = CentroidHom.starCenterToCentroid a := rfl
```
or is the no coercion on `Centroid`?

Reply:
If what you had was correct, you can use `simps` after all here.

Reply:
Sorry - I forgot to modify the statement.